### DISCUSSION TITLE
Welcome to kucoin_arbitrage Discussions!

### DISCUSSION THREAD

**Discussion by kanekoshoyu** [kanekoshoyu/kucoin_arbitrage#6] [Announcements]
I just set up this discussion channel so that you can ask technical questions here and other people can see it.

**Comment by kanekoshoyu** [comment]
Hi Michael,

Thanks for keeping me into the loop, I really appreciate it. I've been way
too busy and stuck at my day job there's not much progress for the
arbitrage bot. But I will soon make a pull request on the serial order
placement, programmable budget via config.ini and some more documentations.
I hope I can get it done this weekend.

Best,
Sho

**Reply by kanekoshoyu** [reply]
Hi @Mihaug,
I have finished a prototype on the sequential order placement, but I still need to fix several things. I've been having problems trying to subscribe to the tradeOrderV2 private websocket channel which publishes a message when an order is placed or filled. Apparently I cannot get those messages, so instead I tried using Balances channel which publishes when there is a change in the trading account currency changes. However, this back and forth seems a bit too slow for the triangular arbitrage to run, there is quite a lot of slippage. I am doing a few more tests on the network latency to see if this issue improves. The upcoming version 0.0.10 will include these updates.

**Reply by kanekoshoyu** [reply]
@Mihaug I fixed the issue with TradeOrder API and now the pipeline is fixed. You can set your budget within the config. I updated a bit of docs so its easier to follow the code. USE AT YOUR OWN RISK, I haven't got enough time to test everything.  

However, the sequential order placement highlights the network latency issue. I did some benchmark with test_kucoin_order_private_latency.rs, and I am seeling around 320ms of latency between before order placement and TradeReceivedMsg reception (order_placement_network). I added a test_network_latency.rs code which prints the ping RTT with the kucoin endpoint, which I am seeing 80-250ms. I will deploy the code into AWS to see how much of it can be reduced, I am aiming the order_placement_network to be less than 30ms.

**Comment by Kinoshnik91** [comment]
Hi!I'm new to programming.Can you tell me how to solve the problem?I'm using VS code.

error[E0432]: unresolved import `tokio::signal::unix`
  --> src\bin\event_triangular.rs:21:20
   |
21 | use tokio::signal::unix::{signal, SignalKind};
   |                    ^^^^ could not find `unix` in `signal`

**Reply by kanekoshoyu** [reply]
I have been building this in both Linux and macOS. Are you using Windows? I suspect you do not have unix stuff in Windows, and that's why it's complaining.

**Comment by TraderHN** [comment]
Hello, can someone help me fix this problem. WARN tokio-runtime-worker kucoin_arbitrage::strategy::all_taker_btc_usd: 48: trying to get from unregistered orderbook [BTC-USDT]

**Reply by kanekoshoyu** [reply]
Hi, I have lately not been updating the repo since I wanted to do a rewrite of some components such that I can properly implement a backtesting module (i.e. this repo is half dead for now). If it is any software design suggestion I would like to hear, but for any debugging I won't answer directly.

**Reply by kanekoshoyu** [reply]
plus if I am not mistaken I saw you posting links to NSFW contents in my discord server which is upsetting. Please behave.
